src/audio/volume/volume.c

```diff
@@ -292,6 +292,27 @@ static void volume_ramp(struct comp_dev *dev)
 	vol_sync_host(dev, cd->channels);
 }
 
+static void reset_state(struct vol_data *cd)
+{
+	int i;
+
+	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++) {
+		cd->volume[i]  =  MAX(MIN(cd->vol_max, VOL_ZERO_DB),
+				      cd->vol_min);
+		cd->tvolume[i] = cd->volume[i];
+		cd->mvolume[i] = cd->volume[i];
+		cd->muted[i] = false;
```

Comment:
oh one thing, could this possibly unmute a channel through a close/open procedure even in control is still muted?

Reply:
That was my worry too. Seem like CI does not run yet sof-test/test-case/check-volume-levels.sh. I just ran the  testmanually and this PR failed while git main branch passes. The kernel driver apparently does not re-apply all the controls if reset state clearing impacts. I think I should remove the volume commit from this PR and study if it's a bug.

Reply:
yes i agree. I think control data is outside the scope of reset. I think its pipeline state that is only supposed to be reset